Classes/Repository/RepositoryClient.swift

```diff
@@ -0,0 +1,152 @@
+//
+//  RepositoryClient.swift
+//  Freetime
+//
+//  Created by Sherlock, James on 30/07/2017.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+import Apollo
+
+protocol RepositoryLoadable {
+    var owner: String { get }
+    var name: String { get }
+    var hasIssuesEnabled: Bool { get }
+}
+
+extension RepositoryLoadable {
+    var nameWithOwner: String {
+        return owner + "/" + name
+    }
+}
+
+protocol RepositoryQuery {
+    init(owner: String, name: String, after: String?)
+    func summaryTypes(from data: GraphQLMappable) -> [IssueSummaryType]?
```

Comment:
Is there a difference between this being empty or `nil`? Feel like you could just make this non-optional?